### ISSUE TITLE
Migrate to Scala 3

### ISSUE THREAD

**Issue by jeffmay** [guizmaii/poc_compile_time_caliban_client_generation#129]
Is it possible to use the code generation with Scala 3.x and ZIO 2.x?

I tried to do this locally and had issues with:

```
[error] (potatoes / update) sbt.librarymanagement.ResolveException: Error downloading com.github.ghostdogpr:caliban-tools_3:2.0.0-RC2
[error]   Not found
[error]   Not found
[error]   not found: /Users/jeffmay/.ivy2/localcom.github.ghostdogpr/caliban-tools_3/2.0.0-RC2/ivys/ivy.xml
[error]   not found: https://repo1.maven.org/maven2/com/github/ghostdogpr/caliban-tools_3/2.0.0-RC2/caliban-tools_3-2.0.0-RC2.pom
```

It looks like the [caliban-tools](https://index.scala-lang.org/ghostdogpr/caliban/artifacts/caliban-tools?pre-releases=true#) project is not compiled to Scala 3, and I'm not sure how the dependency is being added (otherwise I would attempt to use the Scala 2.13 compiled version).

Is this dependent on fixing [this issue](https://github.com/ghostdogpr/caliban/issues/1239)? It looks like a potential fix has been sitting as a [pull request](https://github.com/ghostdogpr/caliban/pull/1248) since the beginning of this year. Is there any work around? Or does this PR need to be merged first?

Thanks!

**Comment by guizmaii** [comment]
Hey,

Sorry for the late reply.

> Is it possible to use the code generation with Scala 3.x and ZIO 2.x?

Not yet but someone is actively working on it here: https://github.com/ghostdogpr/caliban/pull/1415

I'm currently trying to migrate this POC to ZIO2 here: https://github.com/guizmaii/poc_compile_time_caliban_client_generation/pull/155

When it'll be done, I'll migrate it to Scala3 in another PR ;)

**Comment by guizmaii** [comment]
FYI @jeffmay , it has been fixed in v2.0.1 of Caliban: https://github.com/ghostdogpr/caliban/releases/tag/v2.0.1
